src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpNavigationBar.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpNavigationBar : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpNavigationBar(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpNavigationBar))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.NavigationBar)]
+        public void VerifyNavBar()
+        {
+            VisualStudio.Editor.SetText(@"
+class C
+{
+    public void M(int i) { }
+    private C this[int index] { get { return null; } set { } }
+    public static bool operator ==(C c1, C c2) { return true; }
+    public static bool operator !=(C c1, C c2) { return false; }
+}
+
+struct S
+{
+    int Foo() { }
+    void Bar() { }
+}");
+            VisualStudio.Editor.PlaceCaret("this", charsOffset: 1);
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.NavigationBar);
+            VisualStudio.Editor.ExpandRightNavBar();
+            var expectedItems = new[]
+            {
+                "M(int i)",
+                "operator !=(C c1, C c2)",
+                "operator ==(C c1, C c2)",
+                "this[int index]"
+            };
+
+            Assert.Equal(expectedItems, VisualStudio.Editor.GetRightNavBarItems());
+            VisualStudio.Editor.SelectRightNavBarItem("operator !=(C c1, C c2)");
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.NavigationBar);
+
+            VisualStudio.Editor.Verify.CaretPosition(205);
```

Comment:
I think you can do VerifyCurrentLine and include a $$ to denote the caret position instead of setting a huge incomprehensible number.